### PR TITLE
[ERC20 Launchpad] - Endpoint integration for Launch token step

### PR DESCRIPTION
**Pull request by mzparacha**
## Link to Issue
Closes: https://github.com/hicommonwealth/commonwealth/issues/9207

## Description of Changes
Added endpoint integration for Launch token step

## "How We Fixed It"
N/A

## Test Plan
- Enable `FLAG_TOKENIZED_COMMUNITY`
- Visit `/createTokenCommunity`
- Verify you see the auth modal popup with only EVM wallet options (if you are not logged in already)
- Verify you see the address selection modal popup (if you are logged in)
- If you close the auth modal or the address selection modal without authentication, then verify you get these modals when you try to submit.
- Fill the form - verify the form validation works as expected
- Submit - the form submission will fail with this error ATM (plz see "Other Considerations" for more info)
image

## Deployment Plan
N/A

## Other Considerations
- ATM the form submission will throw an error like this, it will be fixed when we deploy the token specific abis (cc: @ianrowan)
image

- There are some `TODO's` left intentionally in the code that relates to adding some token-specific information, these will be done in a followup.

### CODE REVIEW

File: packages/commonwealth/client/scripts/helpers/ContractHelpers/Launchpad.ts [hunk 1]
```diff
@@ -16,6 +17,23 @@ class LaunchpadBondingCurve extends ContractBase {
     this.tokenAddress = tokenAddress;
   }
 
+  async launchToken(name: string, symbol: string, walletAddress: string) {
+    if (!this.initialized || !this.walletEnabled) {
+      await this.initialize(true);
+    }
+
+    const txReceipt = await launchToken(
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/helpers/ContractHelpers/Launchpad.ts:25]: let's not duplicate "launchToken"

File: packages/commonwealth/client/scripts/views/pages/LaunchToken/steps/TokenInformationStep/TokenInformationForm/TokenInformationForm.tsx [hunk 2]
```diff
@@ -25,25 +39,91 @@ import { tokenInformationFormValidationSchema } from './validation';
...
+      // call endpoint
+      const payload = {
+        chainRpc: baseNode.url,
+        ethChainId: baseNode.ethChainId,
+        name: values.tokenName.trim(),
+        symbol: values.tokenTicker.trim(),
+        walletAddress: selectedAddress.address,
+        // TODO 9207: where to store values.tokenDescription and values.tokenImageURL
+      };
+      await launchToken(payload).catch(console.error);
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/pages/LaunchToken/steps/TokenInformationStep/TokenInformationForm/TokenInformationForm.tsx:111]: no try catch ? + some "notifyError" ?

File: packages/commonwealth/client/scripts/views/pages/LaunchToken/steps/TokenInformationStep/TokenInformationForm/TokenInformationForm.tsx [hunk 3]
```diff
@@ -25,25 +39,91 @@ import { tokenInformationFormValidationSchema } from './validation';
...
+
+      onSubmit(values);
+    },
+    [openAddressSelectionModal, selectedAddress, onSubmit, launchToken],
+  );
+
+  useEffect(() => {
+    if (shouldSubmitOnAddressSelection.current) {
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/pages/LaunchToken/steps/TokenInformationStep/TokenInformationForm/TokenInformationForm.tsx:119]: can you explain whats going on here? + maybe add some comment as it looks too complicaed

File: packages/commonwealth/client/scripts/views/pages/LaunchToken/steps/TokenInformationStep/TokenInformationForm/TokenInformationForm.tsx [hunk 4]
```diff
@@ -25,25 +39,91 @@ import { tokenInformationFormValidationSchema } from './validation';
...
+      const nodes = fetchCachedNodes();
+      const baseNode = nodes?.find(
+        (n) => n.ethChainId === commonProtocol.ValidChains.Base,
+      );
+      if (!baseNode || !baseNode.ethChainId) {
+        notifyError('Could not find base chain node');
+        return;
+      }
+
+      // call endpoint
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/pages/LaunchToken/steps/TokenInformationStep/TokenInformationForm/TokenInformationForm.tsx:102]: again, those comments are redundant, makes code look like ai generated and do not have any value for developers

File: packages/commonwealth/client/scripts/views/pages/LaunchToken/steps/TokenInformationStep/TokenInformationForm/TokenInformationForm.tsx (hunk 2 above)
Comment (masvelio) [reply packages/commonwealth/client/scripts/views/pages/LaunchToken/steps/TokenInformationStep/TokenInformationForm/TokenInformationForm.tsx:111]: so we create a token after first step and in second we create a community? Or whats the realtion? What happens if after creating a token, user exit out from the flow without creating a community?

**Review by masvelio — APPROVED** [review]

File: packages/commonwealth/client/scripts/views/pages/LaunchToken/steps/TokenInformationStep/TokenInformationForm/TokenInformationForm.tsx (hunk 4 above)
Comment (mzparacha) [reply packages/commonwealth/client/scripts/views/pages/LaunchToken/steps/TokenInformationStep/TokenInformationForm/TokenInformationForm.tsx:102]: haha sure will replace

Comment (mzparacha) [reply packages/commonwealth/client/scripts/views/pages/LaunchToken/steps/TokenInformationStep/TokenInformationForm/TokenInformationForm.tsx:102]: removed

File: packages/commonwealth/client/scripts/views/pages/LaunchToken/steps/TokenInformationStep/TokenInformationForm/TokenInformationForm.tsx (hunk 3 above)
Comment (mzparacha) [reply packages/commonwealth/client/scripts/views/pages/LaunchToken/steps/TokenInformationStep/TokenInformationForm/TokenInformationForm.tsx:119]: added a comment

File: packages/commonwealth/client/scripts/helpers/ContractHelpers/Launchpad.ts (hunk 1 above)
Comment (mzparacha) [reply packages/commonwealth/client/scripts/helpers/ContractHelpers/Launchpad.ts:25]: do you mean to change the name a bit? - this file actually provides proxy methods to call these contract methods.

Comment (masvelio) [reply packages/commonwealth/client/scripts/helpers/ContractHelpers/Launchpad.ts:25]: yeah like handleLaunchToken or something

**Review by Israellund — APPROVED** [review]
Approving, but agree with Marcin's comments

File: packages/commonwealth/client/scripts/views/pages/LaunchToken/steps/TokenInformationStep/TokenInformationForm/TokenInformationForm.tsx (hunk 2 above)
Comment (mzparacha) [reply packages/commonwealth/client/scripts/views/pages/LaunchToken/steps/TokenInformationStep/TokenInformationForm/TokenInformationForm.tsx:111]: I am doing some updates in #8707 which will fix this.

**Pull request merged**